include/pacmod3_dbc_ros_api.h

```diff
@@ -0,0 +1,189 @@
+#include <pacmod3_msgs/MotorRpt3.h>
+#include <pacmod3_msgs/NotificationCmd.h>
+#include <pacmod3_msgs/OccupancyRpt.h>
+#include <pacmod3_msgs/RearLightsRpt.h>
+#include <pacmod3_msgs/ShiftAuxRpt.h>
+#include <pacmod3_msgs/SteeringAuxRpt.h>
+#include <pacmod3_msgs/SteeringCmd.h>
+#include <pacmod3_msgs/SystemCmdBool.h>
+#include <pacmod3_msgs/SystemCmdFloat.h>
+#include <pacmod3_msgs/SystemCmdInt.h>
+#include <pacmod3_msgs/SystemRptBool.h>
+#include <pacmod3_msgs/SystemRptFloat.h>
+#include <pacmod3_msgs/SystemRptInt.h>
+#include <pacmod3_msgs/TurnAuxRpt.h>
+#include <pacmod3_msgs/VehicleDynamicsRpt.h>
+#include <pacmod3_msgs/VehicleSpeedRpt.h>
+#include <pacmod3_msgs/VinRpt.h>
+#include <pacmod3_msgs/WheelSpeedRpt.h>
+#include <pacmod3_msgs/WiperAuxRpt.h>
+#include <pacmod3_msgs/YawRateRpt.h>
+
+namespace pm_msgs = pacmod3_msgs;
+namespace cn_msgs = can_msgs;
+
+#endif  // ROS1
+
+#if ROS_VERSION==2
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <can_msgs/msg/frame.hpp>
+
+#include <pacmod3_msgs/msg/accel_aux_rpt.hpp>
+#include <pacmod3_msgs/msg/all_system_statuses.hpp>
+#include <pacmod3_msgs/msg/ang_vel_rpt.hpp>
+#include <pacmod3_msgs/msg/brake_aux_rpt.hpp>
+#include <pacmod3_msgs/msg/component_rpt.hpp>
+#include <pacmod3_msgs/msg/date_time_rpt.hpp>
+#include <pacmod3_msgs/msg/detected_object_rpt.hpp>
+#include <pacmod3_msgs/msg/door_rpt.hpp>
+#include <pacmod3_msgs/msg/engine_rpt.hpp>
+#include <pacmod3_msgs/msg/global_cmd.hpp>
+#include <pacmod3_msgs/msg/global_rpt.hpp>
+#include <pacmod3_msgs/msg/headlight_aux_rpt.hpp>
+#include <pacmod3_msgs/msg/interior_lights_rpt.hpp>
+#include <pacmod3_msgs/msg/lat_lon_heading_rpt.hpp>
+#include <pacmod3_msgs/msg/linear_accel_rpt.hpp>
+#include <pacmod3_msgs/msg/motor_rpt1.hpp>
+#include <pacmod3_msgs/msg/motor_rpt2.hpp>
+#include <pacmod3_msgs/msg/motor_rpt3.hpp>
+#include <pacmod3_msgs/msg/notification_cmd.hpp>
+#include <pacmod3_msgs/msg/occupancy_rpt.hpp>
+#include <pacmod3_msgs/msg/rear_lights_rpt.hpp>
+#include <pacmod3_msgs/msg/shift_aux_rpt.hpp>
+#include <pacmod3_msgs/msg/steering_aux_rpt.hpp>
+#include <pacmod3_msgs/msg/steering_cmd.hpp>
+#include <pacmod3_msgs/msg/system_cmd_bool.hpp>
+#include <pacmod3_msgs/msg/system_cmd_float.hpp>
+#include <pacmod3_msgs/msg/system_cmd_int.hpp>
+#include <pacmod3_msgs/msg/system_rpt_bool.hpp>
+#include <pacmod3_msgs/msg/system_rpt_float.hpp>
+#include <pacmod3_msgs/msg/system_rpt_int.hpp>
+#include <pacmod3_msgs/msg/turn_aux_rpt.hpp>
+#include <pacmod3_msgs/msg/vehicle_dynamics_rpt.hpp>
+#include <pacmod3_msgs/msg/vehicle_speed_rpt.hpp>
+#include <pacmod3_msgs/msg/vin_rpt.hpp>
+#include <pacmod3_msgs/msg/wheel_speed_rpt.hpp>
+#include <pacmod3_msgs/msg/wiper_aux_rpt.hpp>
+#include <pacmod3_msgs/msg/yaw_rate_rpt.hpp>
+
+namespace pm_msgs = pacmod3_msgs::msg;
+namespace cn_msgs = can_msgs::msg;
+
+#endif  // ROS2
+
+namespace pacmod3_common
+{
+
+class DbcApi
+{
+public:
+  // Parsing functions take in a ROS CAN msg and return a pointer to a ROS pacmod msg
```

Comment:
```
explicit DbcApi(version = 0):dbc_major_version_(version){
}
```

Reply:
Thanks! I had originally tried something like this without luck. Got it working now with the help of your other comments.